tests/event.test.ts

```diff
@@ -0,0 +1,248 @@
+    const feedback = FeedbackFactory.createEventFeedback(3);
+
+    await new PortalState()
+      .createUsers([user])
+      .createEvents([event])
+      .attendEvents([user], [event], false)
+      .write();
+
+    await ControllerFactory.event(conn).submitEventFeedback({ uuid: event.uuid }, { feedback }, user);
+
+    const attendanceResponse = await ControllerFactory.attendance(conn).getAttendancesForCurrentUser(user);
+
+    expect(attendanceResponse.attendances[0].feedback).toEqual(feedback);
+    expect(user.points).toEqual(event.pointValue + Config.pointReward.EVENT_FEEDBACK_POINT_REWARD);
+  });
+
+  test('is rejected on submission to an event not attended', async () => {
+    const conn = await DatabaseConnection.get();
+    const event = EventFactory.fakeOngoingEvent();
+    const user = UserFactory.fake();
+    const feedback = FeedbackFactory.createEventFeedback(3);
+
+    await new PortalState()
+      .createUsers([user])
+      .createEvents([event])
+      .write();
+
+    await expect(
+      ControllerFactory.event(conn).submitEventFeedback({ uuid: event.uuid }, { feedback }, user),
+    ).rejects.toThrow('You must attend this event before submiting feedback');
+  });
+
+  test('is rejected if submitted to an event multiple times', async () => {
+    const conn = await DatabaseConnection.get();
+    const event = EventFactory.fakeOngoingEvent();
+    const user = UserFactory.fake();
+    const feedback = FeedbackFactory.createEventFeedback(3);
+
+    await new PortalState()
+      .createUsers([user])
+      .createEvents([event])
+      .attendEvents([user], [event], false)
+      .write();
+
+    const eventController = ControllerFactory.event(conn);
+    await eventController.submitEventFeedback({ uuid: event.uuid }, { feedback }, user);
+
+    await expect(
+      eventController.submitEventFeedback({ uuid: event.uuid }, { feedback }, user),
+    ).rejects.toThrow('You cannot submit feedback for this event more than once');
+  });
+
+  test('is rejected if sent after 2 days of event completion', async () => {
+    const conn = await DatabaseConnection.get();
+    const event = EventFactory.fakePastEvent(3);
+    const user = UserFactory.fake();
+    const feedback = FeedbackFactory.createEventFeedback(3);
+
+    await new PortalState()
+      .createUsers([user])
+      .createEvents([event])
+      .attendEvents([user], [event], false)
+      .write();
+
+    await expect(
+      ControllerFactory.event(conn).submitEventFeedback({ uuid: event.uuid }, { feedback }, user),
+    ).rejects.toThrow('You must submit feedback within 2 days of the event ending');
+  });
+
+  test('is rejected if more than 3 feedback is provided', async () => {
+    const feedback = FeedbackFactory.createEventFeedback(4);
+    const errors = await validate(plainToClass(SubmitEventFeedbackRequest, { feedback }));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('feedback');
```

Comment:
Can we add an `expect` that checks that the reason validation failed is because >3?